static/app/components/replays/diff/picker/diffTimestampPicker.tsx

```diff
@@ -0,0 +1,141 @@
+      afterOptions: rrwebFrames
+        .filter(frame => frame.timestamp > timestamp)
+        .slice(0, maxOptions),
+    };
+  }, [frameOrEvent, replay]);
+
+  if (!isHydrateCrumb(frameOrEvent)) {
+    return null;
+  }
+
+  return (
+    <Fragment>
+      <DiffHeader>
+        <Before offset={leftOffsetMs} startTimestampMs={startTimestampMs} />
+        <After offset={rightOffsetMs} startTimestampMs={startTimestampMs} />
+      </DiffHeader>
+      <Wrapper>
+        <List>
+          {beforeOptions.map((item, i) => (
+            <ListItem key={i} data-before="true" data-selected={undefined}>
+              <Tooltip title={t('Select this change as the Before moment')}>
+                <MutationOption
+                  key={i}
+                  frame={item}
+                  startTimestampMs={startTimestampMs}
+                  radioName="left"
+                  onChange={() => {
+                    setLeftOffsetMs(item.timestamp - startTimestampMs);
+                  }}
+                  isChecked={item.timestamp === leftTimestampMs}
+                />
+              </Tooltip>
+            </ListItem>
+          ))}
+        </List>
+
+        <CrumbItem crumb={frameOrEvent} startTimestampMs={startTimestampMs} />
+
+        <List>
+          {afterOptions.map((item, i) => (
+            <ListItem key={i} data-after="true" data-selected={undefined}>
+              <Tooltip title={t('Select this change as the After moment')}>
+                <MutationOption
+                  key={i}
+                  frame={item}
+                  startTimestampMs={startTimestampMs}
+                  radioName="right"
+                  onChange={() => {
+                    setRightOffsetMs(item.timestamp - startTimestampMs);
+                  }}
+                  isChecked={item.timestamp === rightTimestampMs}
+                />
+              </Tooltip>
+            </ListItem>
+          ))}
+        </List>
+      </Wrapper>
+    </Fragment>
+  );
+}
+
+const Wrapper = styled('div')`
+  display: grid;
+  grid-template-columns: max-content ${space(4)} max-content;
+  justify-content: center;
+  margin-top: 1em; /* Reserve space for the CrumbItem title */
+`;
+
+const List = styled('ul')`
+  display: flex;
+  flex-direction: row;
+
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  gap: ${space(1)};
+`;
+
+const ListItem = styled('li')`
+  margin: 0;
+  font-variant-numeric: tabular-nums;
+  background-color: transparent;
+
+  border-radius: ${p => p.theme.borderRadius};
+  border: 1px solid transparent;
+
+  &[data-before='true'] {
+    border-color: ${p => p.theme.red400};
+  }
+  &[data-after='true'] {
+    border-color: ${p => p.theme.green400};
+  }
+
+  &[data-before='true']:hover,
+  &[data-before='true'][data-selected='true'] {
+    background-color: ${p => p.theme.red100};
+  }
+  &[data-after='true']:hover,
+  &[data-after='true'][data-selected='true'] {
+    background-color: ${p => p.theme.green100};
+  }
```

Comment:
are these data attributes only used for styling?

Reply:
yeah, it's instead of doing the typed params thing like `Foo = styled('div')<{after: boolean, before: boolean}>` which generates unique classnames for each combination.